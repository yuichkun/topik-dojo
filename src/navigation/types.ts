/**
 * ナビゲーション型定義
 */

import type { NativeStackScreenProps } from '@react-navigation/native-stack';

// ルートスタックのパラメータ定義
export type RootStackParamList = {
  Top: undefined;
  LearningModeSelection: {
    level: number;
  };
  UnitSelection: { 
    level: number; 
  };
  Learning: { 
    level: number; 
    unitNumber: number; 
  };
  // TODO: 他の画面のパラメータ定義を追加
  // TestModeSelection: { level: number };
  // Results: { level: number };
  // ListeningTest: { level: number; unitId: string };
  // ReadingTest: { level: number; unitId: string };
  // Review: undefined;
};

// 各画面のプロップス型定義
export type TopScreenProps = NativeStackScreenProps<RootStackParamList, 'Top'>;
export type LearningModeSelectionScreenProps = NativeStackScreenProps<RootStackParamList, 'LearningModeSelection'>;
export type UnitSelectionScreenProps = NativeStackScreenProps<RootStackParamList, 'UnitSelection'>;
export type LearningScreenProps = NativeStackScreenProps<RootStackParamList, 'Learning'>;

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}