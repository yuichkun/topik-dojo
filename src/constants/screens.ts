/**
 * 画面名の定数定義
 */

export const SCREEN_NAMES = {
  TOP: 'Top',
  LEARNING_MODE_SELECTION: 'LearningModeSelection',
  UNIT_SELECTION: 'UnitSelection',
  LEARNING: 'Learning',
  TEST_MODE_SELECTION: 'TestModeSelection',
  TEST_UNIT_SELECTION: 'TestUnitSelection',
  READING_TEST: 'ReadingTest',
  LISTENING_TEST: 'ListeningTest',
  REVIEW: 'Review',
  // TODO: 他の画面名を追加
  // RESULTS: 'Results',
} as const;