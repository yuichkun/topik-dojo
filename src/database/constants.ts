// テーブル名の定数定義（型安全性向上）
export enum TableName {
  UNITS = 'units',
  WORDS = 'words',
  SRS_MANAGEMENT = 'srs_management',
  LEARNING_STATUS = 'learning_status',
  TEST_RESULTS = 'test_results',
  TEST_QUESTIONS = 'test_questions',
  REVIEW_HISTORY = 'review_history',
  LEARNING_PROGRESS = 'learning_progress',
}

// データベース設定の定数定義
export const DATABASE_CONFIG = {
  name: 'TopikDojo',
  testName: ':memory:',
} as const;