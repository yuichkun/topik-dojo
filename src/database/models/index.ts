// モデルのエクスポート
export { default as Unit } from './Unit';
export { default as Word } from './Word';
export { default as SrsManagement } from './SrsManagement';
export { default as TestResult } from './TestResult';
export { default as TestQuestion } from './TestQuestion';
export { default as ReviewHistory } from './ReviewHistory';
export { default as LearningProgress } from './LearningProgress';

// モデルクラスの配列（Database初期化時に使用）
import Unit from './Unit';
import Word from './Word';
import SrsManagement from './SrsManagement';
import TestResult from './TestResult';
import TestQuestion from './TestQuestion';
import ReviewHistory from './ReviewHistory';
import LearningProgress from './LearningProgress';

export const modelClasses = [
  Unit,
  Word,
  SrsManagement,
  TestResult,
  TestQuestion,
  ReviewHistory,
  LearningProgress,
];