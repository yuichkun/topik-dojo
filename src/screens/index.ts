/**
 * 画面コンポーネントのエクスポート
 */

export { default as TopScreen } from './TopScreen';
export { default as LearningModeSelectionScreen } from './LearningModeSelectionScreen';
export { default as UnitSelectionScreen } from './UnitSelectionScreen';
export { default as LearningScreen } from './LearningScreen';
export { default as TestModeSelectionScreen } from './TestModeSelectionScreen';
export { default as TestUnitSelectionScreen } from './TestUnitSelectionScreen';
export { default as ReadingTestScreen } from './ReadingTestScreen';
export { default as ListeningTestScreen } from './ListeningTestScreen';
export { default as ReviewScreen } from './ReviewScreen';