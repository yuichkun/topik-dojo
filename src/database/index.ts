import { Database } from '@nozbe/watermelondb';
import SQLiteAdapter from '@nozbe/watermelondb/adapters/sqlite';

import schema from './schema';
import migrations from './migrations';
import { modelClasses } from './models';
import { DATABASE_CONFIG } from './constants';

// SQLiteアダプターの設定
const adapter = new SQLiteAdapter({
  schema,
  migrations,
  jsi: true, // JSI (JavaScript Interface) を有効化（パフォーマンス向上）
  dbName: DATABASE_CONFIG.name, // データベースファイル名を設定から取得（.dbは自動付与される）
});

// データベースインスタンスの作成
const database = new Database({
  adapter,
  modelClasses,
});

export default database;