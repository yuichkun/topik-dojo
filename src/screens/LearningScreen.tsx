import React, { useState, useEffect } from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  Alert,
  ScrollView,
  ActivityIndicator,
} from 'react-native';
import { playAudio } from '../utils/audioPlayer';
import { LearningScreenProps } from '../navigation/types';
import { Word, SrsManagement } from '../database/models';
import { getWordsByUnit } from '../database/queries/unitQueries';
import {
  getSrsManagementByWordId,
  createSrsManagement,
  calculateDaysToReview,
} from '../database/queries/srsQueries';

export default function LearningScreen({
  route,
  navigation,
}: LearningScreenProps) {
  const { level, unitNumber } = route.params;

  // 状態管理
  const [words, setWords] = useState<Word[]>([]);
  const [currentIndex, setCurrentIndex] = useState(0);
  const [showMeaning, setShowMeaning] = useState(false);
  const [showExample, setShowExample] = useState(false);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);
  const [srsData, setSrsData] = useState<Map<string, SrsManagement>>(new Map());

  // 単語データ取得
  useEffect(() => {
    const loadWords = async () => {
      try {
        setLoading(true);
        const wordsData = await getWordsByUnit(level, unitNumber);

        if (wordsData.length === 0) {
          setError('単語データの読み込みに失敗しました');
          return;
        }

        setWords(wordsData);
        setError(null);

        // 各単語のSRS状況を取得
        const srsMap = new Map<string, SrsManagement>();
        for (const word of wordsData) {
          const srs = await getSrsManagementByWordId(word.id);
          if (srs) {
            srsMap.set(word.id, srs);
          }
        }
        setSrsData(srsMap);
      } catch (err) {
        console.error('単語データ取得エラー:', err);
        setError('単語データの読み込みに失敗しました');
      } finally {
        setLoading(false);
      }
    };

    loadWords();
  }, [level, unitNumber]);

  // 現在の単語
  const currentWord = words[currentIndex];

  // 次の単語へ
  const handleNext = () => {
    if (currentIndex < words.length - 1) {
      setCurrentIndex(currentIndex + 1);
      setShowMeaning(false);
      setShowExample(false);
    } else {
      // 学習完了
      Alert.alert(
        '学習完了',
        `${level}級 ユニット${getUnitRange()} の学習が完了しました。`,
        [
          {
            text: 'OK',
            onPress: () => navigation.goBack(),
          },
        ],
      );
    }
  };

  // 前の単語へ
  const handlePrevious = () => {
    if (currentIndex > 0) {
      setCurrentIndex(currentIndex - 1);
      setShowMeaning(false);
      setShowExample(false);
    }
  };

  // 復習に追加
  const handleMarkForReview = async () => {
    if (currentWord) {
      try {
        const existingSrs = srsData.get(currentWord.id);
        if (!existingSrs) {
          // 新規登録
          const newSrs = await createSrsManagement(currentWord.id, false);
          if (newSrs) {
            const newSrsData = new Map(srsData);
            newSrsData.set(currentWord.id, newSrs);
            setSrsData(newSrsData);
            Alert.alert('復習登録', '復習リストに追加しました');
          }
        }
      } catch (err) {
        console.error('復習登録エラー:', err);
        Alert.alert('エラー', '復習登録に失敗しました');
      }
    }
  };

  // 意味表示切り替え
  const toggleMeaning = () => {
    setShowMeaning(!showMeaning);
  };

  // 例文表示切り替え
  const toggleExample = () => {
    setShowExample(!showExample);
  };

  // 音声再生
  const playWordAudio = async () => {
    if (currentWord) {
      try {
        await playAudio(currentWord.korean, 'word');
      } catch (_error) {
        Alert.alert('エラー', '音声の再生に失敗しました');
      }
    }
  };

  // 例文音声再生
  const playExampleAudio = async () => {
    if (currentWord && currentWord.exampleKorean) {
      try {
        // Use the korean word as key, not the example sentence
        await playAudio(currentWord.korean, 'example');
      } catch (_error) {
        Alert.alert('エラー', '例文音声の再生に失敗しました');
      }
    }
  };

  // ユニット範囲の計算
  const getUnitRange = () => {
    const start = (unitNumber - 1) * 10 + 1;
    const end = unitNumber * 10;
    return `${start}-${end}`;
  };

  // 例文内の単語をハイライト
  const highlightWordInExample = (example: string, word: string) => {
    const wordIndex = example.indexOf(word);
    if (wordIndex === -1) return { before: example, highlighted: '', after: '' };
    
    const before = example.substring(0, wordIndex);
    const highlighted = word;
    const after = example.substring(wordIndex + word.length);
    
    return { before, highlighted, after };
  };

  if (loading) {
    return (
      <View className="flex-1 justify-center items-center bg-white">
        <ActivityIndicator size="large" color="#3B82F6" />
        <Text className="mt-4 text-gray-600">単語を読み込み中...</Text>
      </View>
    );
  }

  if (error) {
    return (
      <View className="flex-1 justify-center items-center bg-white px-4">
        <Text className="text-red-500 text-center mb-4">{error}</Text>
        <TouchableOpacity
          className="bg-blue-500 px-6 py-3 rounded-lg"
          onPress={() => navigation.goBack()}
        >
          <Text className="text-white font-semibold">戻る</Text>
        </TouchableOpacity>
      </View>
    );
  }

  if (!currentWord) {
    return (
      <View className="flex-1 justify-center items-center bg-white">
        <Text className="text-gray-600">単語データがありません</Text>
      </View>
    );
  }

  return (
    <View className="flex-1 bg-white">
      {/* ヘッダー */}
      <View className="bg-blue-500 px-4 py-3 pt-12">
        <View className="flex-row justify-between items-center">
          <TouchableOpacity
            onPress={() => navigation.goBack()}
            className="bg-white/20 px-3 py-2 rounded-lg"
          >
            <Text className="text-white font-semibold">← 戻る</Text>
          </TouchableOpacity>

          <View className="flex-1 mx-4">
            <Text className="text-white text-center font-bold text-lg">
              {level}級 ユニット{getUnitRange()}
            </Text>
            <Text className="text-white/80 text-center">
              ({currentIndex + 1}/{words.length})
            </Text>
          </View>

          <View className="w-16" />
        </View>

        {/* 進捗バー */}
        <View className="bg-white/20 h-2 rounded-full mt-3">
          <View
            className="bg-white h-full rounded-full"
            style={{ width: `${((currentIndex + 1) / words.length) * 100}%` }}
          />
        </View>
      </View>

      {/* メインコンテンツ */}
      <ScrollView className="flex-1 px-6 py-8">
        {/* 単語カード */}
        <View className="bg-gray-50 rounded-xl p-6 mb-6">
          {/* 韓国語単語 */}
          <TouchableOpacity
            onPress={toggleMeaning}
            className="items-center mb-4"
          >
            <Text className="text-4xl font-bold text-gray-800 mb-2">
              {currentWord.korean}
            </Text>

            {/* 音声再生ボタン */}
            <TouchableOpacity
              onPress={playWordAudio}
              className="bg-blue-500 px-4 py-2 rounded-full"
            >
              <Text className="text-white font-semibold">🔊 音声再生</Text>
            </TouchableOpacity>
          </TouchableOpacity>

          {/* 日本語訳 */}
          <TouchableOpacity onPress={toggleMeaning} className="mb-4">
            {showMeaning ? (
              <Text className="text-xl text-center text-gray-700 bg-white p-4 rounded-lg">
                {currentWord.japanese}
              </Text>
            ) : (
              <Text className="text-gray-500 text-center p-4 border-2 border-dashed border-gray-300 rounded-lg">
                タップして意味を表示
              </Text>
            )}
          </TouchableOpacity>

          {/* 例文セクション */}
          {(currentWord.exampleKorean || currentWord.exampleJapanese) && (
            <View>
              <TouchableOpacity
                onPress={toggleExample}
                className="bg-blue-100 px-4 py-2 rounded-lg mb-3"
              >
                <Text className="text-blue-700 font-semibold text-center">
                  {showExample ? '例文を隠す' : '例文を見る'}
                </Text>
              </TouchableOpacity>

              {showExample && (
                <View className="bg-white p-4 rounded-lg space-y-3">
                  {currentWord.exampleKorean && (
                    <View>
                      <View className="flex-row justify-between items-center mb-1">
                        <Text className="text-gray-600 text-sm">
                          韓国語例文
                        </Text>
                        <TouchableOpacity
                          onPress={playExampleAudio}
                          className="bg-blue-500 px-2 py-1 rounded"
                        >
                          <Text className="text-white text-xs">🔊</Text>
                        </TouchableOpacity>
                      </View>
                      <Text className="text-lg text-gray-800 mb-3">
                        {(() => {
                          const parts = highlightWordInExample(
                            currentWord.exampleKorean,
                            currentWord.korean
                          );
                          return (
                            <>
                              {parts.before}
                              <Text className="bg-yellow-200 font-semibold underline">
                                {parts.highlighted}
                              </Text>
                              {parts.after}
                            </>
                          );
                        })()}
                      </Text>
                    </View>
                  )}

                  {currentWord.exampleJapanese && (
                    <View>
                      <Text className="text-gray-600 text-sm mb-1">
                        日本語訳
                      </Text>
                      <Text className="text-lg text-gray-700">
                        {currentWord.exampleJapanese}
                      </Text>
                    </View>
                  )}
                </View>
              )}
            </View>
          )}
        </View>
      </ScrollView>

      {/* フッター操作ボタン */}
      <View className="px-6 py-4 bg-white border-t border-gray-200">
        <View className="flex-row justify-between items-center">
          {/* 前へボタン */}
          <TouchableOpacity
            onPress={handlePrevious}
            disabled={currentIndex === 0}
            className={`px-6 py-3 rounded-lg ${
              currentIndex === 0 ? 'bg-gray-300' : 'bg-gray-500'
            }`}
          >
            <Text
              className={`font-semibold ${
                currentIndex === 0 ? 'text-gray-500' : 'text-white'
              }`}
            >
              前へ
            </Text>
          </TouchableOpacity>

          {/* 復習ボタン/復習予定表示 */}
          {(() => {
            const existingSrs = srsData.get(currentWord.id);
            if (existingSrs) {
              const daysToReview = calculateDaysToReview(
                existingSrs.nextReviewDate || 0,
              );
              return (
                <View className="px-6 py-3 rounded-lg bg-gray-200">
                  <Text className="font-semibold text-gray-700 text-center">
                    {daysToReview === 0
                      ? '今日復習予定'
                      : `${daysToReview}日後に復習予定`}
                  </Text>
                </View>
              );
            } else {
              return (
                <TouchableOpacity
                  onPress={handleMarkForReview}
                  className="px-6 py-3 rounded-lg bg-orange-200"
                >
                  <Text className="font-semibold text-orange-700">
                    復習に追加
                  </Text>
                </TouchableOpacity>
              );
            }
          })()}

          {/* 次へボタン */}
          <TouchableOpacity
            onPress={handleNext}
            className="bg-blue-500 px-6 py-3 rounded-lg"
          >
            <Text className="text-white font-semibold">
              {currentIndex === words.length - 1 ? '完了' : '次へ'}
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}
