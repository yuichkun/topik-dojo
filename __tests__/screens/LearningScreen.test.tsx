/**
 * @format
 */

import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { Alert, TouchableOpacity, Text } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import LearningScreen from '../../src/screens/LearningScreen';
import { RootStackParamList } from '../../src/navigation/types';
import { SCREEN_NAMES } from '../../src/constants/screens';
import { createTestWord } from '../helpers/databaseHelpers';
import { Unit } from '../../src/database/models';
import { TableName } from '../../src/database/constants';
import database from '../../src/database';

// react-native-sound-playerのモック
jest.mock('react-native-sound-player', () => ({
  playAsset: jest.fn(),
  stop: jest.fn(),
  pause: jest.fn(),
  resume: jest.fn(),
}));

// AlertのスパイFunction
jest.spyOn(Alert, 'alert');

const Stack = createNativeStackNavigator<RootStackParamList>();

// テスト用のナビゲーションコンテナーを作成
const createTestNavigationContainer = (level: number, unitNumber: number) => {
  const TestNavigationContainer = () => (
    <NavigationContainer>
      <Stack.Navigator initialRouteName={SCREEN_NAMES.LEARNING}>
        <Stack.Screen
          name={SCREEN_NAMES.LEARNING}
          component={LearningScreen}
          initialParams={{ level, unitNumber }}
          options={{ headerShown: false }}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
  return TestNavigationContainer;
};

describe('LearningScreen', () => {
  beforeEach(async () => {
    jest.clearAllMocks();

    // テスト用ユニットを作成
    await database.write(async () => {
      await database.collections.get<Unit>(TableName.UNITS).create(unit => {
        unit._raw.id = 'unit_1_1';
        unit.grade = 1;
        unit.unitNumber = 1;
      });
    });

    // テスト用単語を作成
    await createTestWord(database, {
      id: 'word_1',
      korean: '테스트',
      japanese: 'テスト',
      exampleKorean: '이것은 테스트입니다.',
      exampleJapanese: 'これはテストです。',
      grade: 1,
      unitId: 'unit_1_1',
      unitOrder: 1,
    });
  });

  test('renders correctly with level 1 unit 1', async () => {
    const TestContainer = createTestNavigationContainer(1, 1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    expect(component!).toBeDefined();
  });

  test('displays Korean word when data is loaded', async () => {
    const TestContainer = createTestNavigationContainer(1, 1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // 韓国語単語が表示されることを確認
    const textElements = component!.root.findAllByType(Text);
    const koreanWordText = textElements.find(element => {
      const children = element.props.children;
      return typeof children === 'string' && children === '테스트';
    });
    expect(koreanWordText).toBeDefined();
  });

  test('displays audio play button when word is loaded', async () => {
    const TestContainer = createTestNavigationContainer(1, 1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // 音声再生ボタンを確認
    const audioButton = component!.root.findByProps({
      children: '🔊 音声再生',
    });
    expect(audioButton).toBeDefined();
  });

  test('audio button can be pressed', async () => {
    const TestContainer = createTestNavigationContainer(1, 1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // TouchableOpacityコンポーネントの中から音声再生ボタンを探す
    const touchableOpacities = component!.root.findAllByType(TouchableOpacity);
    const audioButton = touchableOpacities.find(button => {
      try {
        button.findByProps({ children: '🔊 音声再生' });
        return true;
      } catch {
        return false;
      }
    });

    expect(audioButton).toBeDefined();
    expect(audioButton!.props.onPress).toBeInstanceOf(Function);
  });

  test('displays meaning toggle button', async () => {
    const TestContainer = createTestNavigationContainer(1, 1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // 意味表示切り替えボタンを確認
    const meaningToggle = component!.root.findByProps({
      children: 'タップして意味を表示',
    });
    expect(meaningToggle).toBeDefined();
  });

  test('displays navigation buttons', async () => {
    const TestContainer = createTestNavigationContainer(1, 1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // ナビゲーションボタンを確認
    const prevButton = component!.root.findByProps({ children: '前へ' });
    const completeButton = component!.root.findByProps({ children: '完了' }); // 単語が1つなので「完了」
    const reviewButton = component!.root.findByProps({
      children: '復習に追加',
    });

    expect(prevButton).toBeDefined();
    expect(completeButton).toBeDefined();
    expect(reviewButton).toBeDefined();
  });

  test('displays back button in header', async () => {
    const TestContainer = createTestNavigationContainer(1, 1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    const backButton = component!.root.findByProps({ children: '← 戻る' });
    expect(backButton).toBeDefined();
  });

  test('matches snapshot', async () => {
    const TestContainer = createTestNavigationContainer(1, 1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    expect(component!.toJSON()).toMatchSnapshot();
  });
});
