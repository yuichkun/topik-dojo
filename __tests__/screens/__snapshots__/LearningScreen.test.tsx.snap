// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningScreen matches snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      activityState={2}
      aria-hidden={false}
      collapsable={false}
      fullScreenSwipeShadowEnabled={true}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onHeaderHeightChange={[Function]}
      onNativeDismissCancelled={[Function]}
      onSheetDetentChanged={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents={
        [
          1,
        ]
      }
      sheetCornerRadius={-1}
      sheetElevation={24}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetInitialDetent={0}
      sheetInitialDetentIndex={0}
      sheetLargestUndimmedDetent={-1}
      sheetLargestUndimmedDetentIndex={-1}
      stackPresentation="push"
      style={
        [
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ],
          {
            "zIndex": undefined,
          },
        ]
      }
      swipeDirection="horizontal"
    >
      <RNSScreenContentWrapper
        collapsable={false}
        style={
          [
            {
              "flex": 1,
            },
            [
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          className="flex-1 bg-white"
        >
          <View
            className="bg-blue-500 px-4 py-3 pt-12"
          >
            <View
              className="flex-row justify-between items-center"
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  className="text-white font-semibold"
                >
                  ‚Üê Êàª„Çã
                </Text>
              </View>
              <View
                className="flex-1 mx-4"
              >
                <Text
                  className="text-white text-center font-bold text-lg"
                >
                  1
                  Á¥ö „É¶„Éã„ÉÉ„Éà
                  1-10
                </Text>
                <Text
                  className="text-white/80 text-center"
                >
                  (
                  1
                  /
                  1
                  )
                </Text>
              </View>
              <View
                className="w-16"
              />
            </View>
            <View
              className="bg-white/20 h-2 rounded-full mt-3"
            >
              <View
                className="bg-white h-full rounded-full"
                style={
                  {
                    "width": "100%",
                  }
                }
              />
            </View>
          </View>
          <RCTScrollView
            className="flex-1 px-6 py-8"
          >
            <View>
              <View
                className="bg-gray-50 rounded-xl p-6 mb-6"
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    className="text-4xl font-bold text-gray-800 mb-2"
                  >
                    ÌÖåÏä§Ìä∏
                  </Text>
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      className="text-white font-semibold"
                    >
                      üîä Èü≥Â£∞ÂÜçÁîü
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    className="text-gray-500 text-center p-4 border-2 border-dashed border-gray-300 rounded-lg"
                  >
                    „Çø„ÉÉ„Éó„Åó„Å¶ÊÑèÂë≥„ÇíË°®Á§∫
                  </Text>
                </View>
                <View>
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      className="text-blue-700 font-semibold text-center"
                    >
                      ‰æãÊñá„ÇíË¶ã„Çã
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
          <View
            className="px-6 py-4 bg-white border-t border-gray-200"
          >
            <View
              className="flex-row justify-between items-center"
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  className="font-semibold text-gray-500"
                >
                  Ââç„Å∏
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  className="font-semibold text-orange-700"
                >
                  Âæ©Áøí„Å´ËøΩÂä†
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  className="text-white font-semibold"
                >
                  ÂÆå‰∫Ü
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RNSScreenContentWrapper>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleFontFamily="System"
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleFontFamily="System"
        largeTitleFontWeight="700"
        largeTitleHideShadow={false}
        pointerEvents="box-none"
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "position": "absolute",
            "width": "100%",
          }
        }
        title="Learning"
        titleColor="rgb(28, 28, 30)"
        titleFontFamily="System"
        titleFontWeight="600"
        topInsetEnabled={true}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
